packages/components/src/tools-panel/tools-panel-header/component.tsx

```diff
@@ -199,18 +199,27 @@ const ToolsPanelHeader = (
 						isSmall: true,
 						describedBy: dropdownMenuDescriptionText,
 					} }
+					popoverProps={ {
+						placement: 'left-start',
+						offset: 258, // sidebar width (280px) - button width (24px) + border (2px)
```

Comment:
`useViewportMatch` can be used to detect when the viewport is mobile/desktop and set this value appropriately. 

The bigger issue is that this change is being made to the component in the generic components package, so it's a very opinionated default. If someone ever wanted to implement this component somewhere else in the UI, the offset might not be appropriate.

Probably the most pragmatic option right now is to pass though a `dropdownMenuOffset` prop to every tools panel, that way the offset value can be defined in the block editor package where we know the tools panel exists in the right-hand sidebar (and we should also double-check it works ok for RTL locales).

Reply:
Could possibly instead pass in a `dropdownMenuProps` component to tools panel for better customization.

I've requested a review from a couple of the components package contributors for their thoughts, they will have better ideas on how to achieve this 😄

Reply:
As @talldan mentioned, we definitely don't want to bake sidebar-related logic into the `ToolsPanel` component.

Adding a `dropdownMenuProps` prop to `ToolsPanel` seems the most flexible solution

Reply:
Thanks team, I updated the PR to use a `dropdownMenuProps` prop passed from `ToolsPanel`. Mind taking another look on that implementation? I'll follow-up with `useViewportMatch`.